src/modules/customers/customers.service.ts

```diff
@@ -22,4 +37,120 @@ export class CustomersService {
+    };
+
+    try {
+      const existingCustomersInInformatica = await this.informaticaService.getCustomers(backendQuery);
+      if (existingCustomersInInformatica?.[0]) {
+        if (existingCustomersInInformatica[0]?.isLegacyRecord === false) {
+          res.status(200).json(
+            existingCustomersInInformatica.map(
+              (customerInInformatica): GetCustomersResponseItem => ({
+                partyUrn: customerInInformatica.partyUrn,
+                name: customerInInformatica.name,
+                sfId: customerInInformatica.sfId,
+                companyRegNo: customerInInformatica.companyRegNo,
+                type: customerInInformatica.type,
+                subtype: customerInInformatica.subtype,
+                isLegacyRecord: customerInInformatica.isLegacyRecord,
+              }),
+            ),
+          );
+          return;
+        } else if (existingCustomersInInformatica[0]?.isLegacyRecord === true && existingCustomersInInformatica[0]?.partyUrn) {
+          let dunsNumber: string = null;
+          try {
+            dunsNumber = await this.dunAndBradstreetService.getDunAndBradstreetNumberByRegistrationNumber(DTFSCustomerDto.companyRegistrationNumber);
+          } catch (error) {
+            throw error;
+          }
+          try {
+            const partyUrn = existingCustomersInInformatica[0].partyUrn;
+            const isLegacyRecord = true;
+            const createdCustomer = await this.createCustomerByURNAndDUNS(DTFSCustomerDto, partyUrn, dunsNumber, isLegacyRecord);
+            res.status(201).json(createdCustomer);
+            return;
+          } catch (error) {
+            throw error;
+          }
+        }
+      } else {
+        throw new InternalServerErrorException();
+      }
+    } catch (error) {
+      if (error instanceof NotFoundException) {
+        const createUkefIdDto: CreateUkefIdDto[] = [
+          {
+            numberTypeId: 2,
+            createdBy: 'DTFS Automated Customer Creation User',
+            requestingSystem: 'MDM',
+          },
+        ];
+        // TODO: replace this with a call to Salesforce's NUMGEN table once that's in place
+        let dunsNumber: string = null;
+        try {
+          dunsNumber = await this.dunAndBradstreetService.getDunAndBradstreetNumberByRegistrationNumber(DTFSCustomerDto.companyRegistrationNumber);
+        } catch (error) {
+          throw error;
+        }
+        let partyUrn: string = null;
+        try {
+          const numbersServiceResponse: UkefId[] = await this.numbersService.create(createUkefIdDto);
+          partyUrn = numbersServiceResponse[0].maskedId;
+        } catch (error) {}
+        try {
+          const isLegacyRecord = false;
+          const createdCustomer = await this.createCustomerByURNAndDUNS(DTFSCustomerDto, partyUrn, dunsNumber, isLegacyRecord);
+          res.status(201).json(createdCustomer);
+          return;
+        } catch (error) {
```

Comment:
Same question about `catch with immediate throw`. Maybe this `try catch` is unnecessary? or does it provide any benefit?